cmd/object-handlers-common_test.go

```diff
@@ -51,3 +57,64 @@ func TestCanonicalizeETag(t *testing.T) {
 		}
 	}
 }
+
+// Tests - CheckPreconditions()
+func TestCheckPreconditions(t *testing.T) {
+	objModTime := time.Date(2024, 8, 26, 02, 01, 01, 0, time.UTC)
+	t.Run("If-None-Match", func(t *testing.T) {
+		testCases := []struct {
+			ifNoneMatch     string
+			ifModifiedSince string
+			objInfo         ObjectInfo
+			expectedFlag    bool
+			expectedCode    int
+		}{
+			// If-None-Match(false) and If-Modified-Since(true)
+			{ifNoneMatch: "aa", ifModifiedSince: "Sun, 26 Aug 2024 02:01:00 GMT", objInfo: ObjectInfo{ETag: "aa", ModTime: objModTime}, expectedFlag: true, expectedCode: 304},
+			// If-Modified-Since(false)
+			{ifNoneMatch: "aaa", ifModifiedSince: "Sun, 26 Aug 2024 02:01:01 GMT", objInfo: ObjectInfo{ETag: "aa", ModTime: objModTime}, expectedFlag: true, expectedCode: 304},
+			{ifNoneMatch: "aaa", ifModifiedSince: "Sun, 26 Aug 2024 02:01:02 GMT", objInfo: ObjectInfo{ETag: "aa", ModTime: objModTime}, expectedFlag: true, expectedCode: 304},
+		}
+		for i, tc := range testCases {
+			recorder := httptest.NewRecorder()
+			request := httptest.NewRequest(http.MethodHead, "/bucket/a", bytes.NewReader([]byte{}))
+			request.Header.Set(xhttp.IfNoneMatch, tc.ifNoneMatch)
+			request.Header.Set(xhttp.IfModifiedSince, tc.ifModifiedSince)
+			actualFlag := checkPreconditions(context.Background(), recorder, request, tc.objInfo, ObjectOptions{})
+			if tc.expectedFlag != actualFlag {
+				t.Errorf("Test Name:%s, test case:%d is fail", t.Name(), i+1)
```

Comment:
Rather than call t.Name(), give each testCase a `name string` and then use that, or just use the integer `i` here to identify which case has failed. Always provide the "got" and "want" to know why the case is failing.

```suggestion
				t.Errorf("test: %s, got flag: %s, want: %s", tc.name, actualFlag, tc.expectedFlag)
```

Reply:
okay, thanks

Reply:
actually even better use `t.Run("", func(t *testing.T) {})`